devel/build-to-localhost

```diff
@@ -46,8 +47,8 @@ docker buildx build \
     --cache-from $BASE_BUILDER_IMAGE \
     --cache-from $BASE_IMAGE \
     --cache-to type=inline \
-    --tag $BASE_BUILDER_IMAGE:$tag \
-    --load \
+    --tag localhost:5000/$BASE_BUILDER_IMAGE:$tag \
+    --push \
```

Comment:
Since this means the built images are no longer available for use locally, maybe a script to call `docker pull localhost:5000/nextstrain/base:$tag` similar to the existing [`./devel/pull`](https://github.com/nextstrain/docker-base/blob/c3857f55612e4ccc5834348ea51a9c57acb7f9f7/devel/pull) would be reasonable to add alongside these changes.

Reply:
Also noting that I don't think `./devel/pull` is useful in CI, though it's still good to keep around for local dev. With single-platform builds it is redundant: `docker buildx build --cache-from` should pull if it doesn't exist locally yet. With multi-platform builds it is redundant and incomplete: unless specifying `--platform` and retagging to avoid overwrite, `pull` will not keep layers for all platforms locally.

Reply:
Nod. What's the use case for it in local dev then?

Reply:
Hm, actually probably not useful for local dev either. I previously thought that `--cache-from` would not pull from the registry if an image with the same tag is available locally, but [the docs](https://docs.docker.com/engine/reference/commandline/buildx_build/#cache-from) say otherwise:

> Use an external cache source for a build (--cache-from)
> If no type is specified, `registry` exporter is used with a specified reference.

So I think we can just get rid of `./devel/pull` independent from the changes in this PR.

Reply:
Sounds good.

Reply:
#87

Reply:
> maybe a script to call `docker pull localhost:5000/nextstrain/base:$tag` similar to the existing [`./devel/pull`](https://github.com/nextstrain/docker-base/blob/c3857f55612e4ccc5834348ea51a9c57acb7f9f7/devel/pull) would be reasonable to add alongside these changes.

Done with 31ef757733c939ac7d801cbd81d628f4aaf38993.